pkg/controllers/nodeclaim/disruption/expiration.go

```diff
@@ -51,19 +49,16 @@ func (e *Expiration) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool,
 		}
 		return reconcile.Result{}, nil
 	}
-	node, err := nodeclaimutil.NodeForNodeClaim(ctx, e.kubeClient, nodeClaim)
-	if nodeclaimutil.IgnoreNodeNotFoundError(nodeclaimutil.IgnoreDuplicateNodeError(err)) != nil {
-		return reconcile.Result{}, err
-	}
-	// We do the expiration check in this way since there is still a migration path for creating nodeclaims from Nodes
-	// In this case, we need to make sure that we take the older of the two for expiration
-	var expirationTime time.Time
-	if node == nil {
-		expirationTime = nodeClaim.CreationTimestamp.Add(*nodePool.Spec.Disruption.ExpireAfter.Duration)
-	} else {
-		expirationTime = node.CreationTimestamp.Add(*nodePool.Spec.Disruption.ExpireAfter.Duration)
+	// 2. If NodeClaim is not launched, remove the expired status condition
+	if launchCond := nodeClaim.StatusConditions().GetCondition(v1beta1.Launched); launchCond == nil || launchCond.IsFalse() {
```

Comment:
Why do we care? Shouldn't we expire the node regardless if it is launched or not?